src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -491,28 +491,39 @@ CHIP_ERROR OTARequestor::SendQueryImageRequest(OperationalDeviceProxy & devicePr
     constexpr OTADownloadProtocol kProtocolsSupported[] = { OTADownloadProtocol::kBDXSynchronous };
     QueryImage::Type args;
 
-    VerifyOrReturnError(Basic::Attributes::VendorID::Get(kRootEndpointId, &args.vendorId) == EMBER_ZCL_STATUS_SUCCESS,
-                        CHIP_ERROR_READ_FAILED);
+    uint16_t vendorId;
+    VerifyOrReturnError(ConfigurationMgr().GetVendorId(vendorId) == CHIP_NO_ERROR, CHIP_ERROR_READ_FAILED);
+    args.vendorId = static_cast<VendorId>(vendorId);
```

Comment:
@bzbarsky-apple For software version, the type in ConfigurationMgr was incorrectly `uint16_t`. Since it was used way less, it was much easier for me update that type to uint32_t. For vendorID, I initially started to modify from `uint16_t` -> `VendorId` but it is used in many places and would result in modifying way more code than I want to just to make the type match with the spec. I opted for the cast here. If it's super important, let me know and I can create a separate PR (or at least file an issue) to make that change.

Reply:
I think this is fine.  Keeps the scope creep down.  I do think an issue is worth filing, but fixing that seems somewhat low-priority.

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/13877 for post v1.0 work.